### PR TITLE
Add torch.distributed get_rank and get_world_size to constant_fold_functions

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109029

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/variables/torch.py [hunk 1]
```diff
@@ -82,6 +82,8 @@
     torch.is_floating_point,
     torch.nn.functional._Reduction.get_enum,
     torch._C._get_privateuse1_backend_name,
+    torch.distributed.get_rank,
+    torch.distributed.get_world_size,
```
Comment (bdhirsh) [inline torch/_dynamo/variables/torch.py:86]: basic Q: it seems a bit redundant to force dynamo to inline a hardcoded list of torch API's into the graph, if we know that AOTAutograd will do it later anyway. Is main purpose to (mildly) limit the API surface for any backends that want to do stuff before AOTAutograd?

Comment (ezyang) [reply torch/_dynamo/variables/torch.py:86]: It's actually not inlining; this says that it's OK to run the operation, get out a Python constant, and treat it as constant for the purposes of the rest of Dynamo execution

Comment (bdhirsh) [reply torch/_dynamo/variables/torch.py:86]: ah got it. thanks!

**Review by bdhirsh — APPROVED** [review]

**Pull request closed without merging**